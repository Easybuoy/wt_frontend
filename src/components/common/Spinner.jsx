import React from "react";
import { Spinner, Box } from "@chakra-ui/core";
import PropTypes from "prop-types";

function CustomSpinner({ thickness, size, text }) {
  return (
    <Box>
      <Spinner
        thickness={thickness}
        emptyColor="gray.200"
        color="orange.400"
        size={size}
      />
      {text ? <Box ml={3}>{text}</Box> : null}
    </Box>
  );
}

//adding proptypes
CustomSpinner.propTypes = {
  thickness: PropTypes.string.isRequired,
  size: PropTypes.string.isRequired,
  text: PropTypes.string
};

export default CustomSpinner;
