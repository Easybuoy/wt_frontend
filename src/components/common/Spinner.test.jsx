import React from "react";
import { shallow } from "enzyme";

import Spinner from "./Spinner";

describe("<Spinner />", () => {
  let wrapper;

  const props = {
    thickness: "thick",
    size: "1rem"
  };

  beforeEach(() => {
    wrapper = shallow(<Spinner {...props} />);
  });

  it("renders the Spinner component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
