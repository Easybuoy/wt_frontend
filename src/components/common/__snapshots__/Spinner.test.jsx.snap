// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders the Spinner component correctly 1`] = `
<Box>
  <Spinner
    color="orange.400"
    emptyColor="gray.200"
    size="1rem"
    thickness="thick"
  />
</Box>
`;
